internal/rbd/rbd_util.go

```diff
@@ -426,9 +426,15 @@ func flattenClonedRbdImages(ctx context.Context, snaps []snapshotInfo, pool, mon
 		util.ErrorLog(ctx, "failed to open connection %s; err %v", rv, err)
 		return err
 	}
-	for _, s := range snaps {
-		if s.Namespace.Type == "trash" {
-			rv.RbdImageName = s.Namespace.OriginalName
+
+	for _, snapInfo := range snaps {
+		// get original snap name for the snapshots in trash namespace
+		origSnapName, err := rv.getOrigSnapName(snapInfo.Id)
+		if err != nil {
+			return err
+		}
+		if origSnapName != "" {
```

Comment:
when origSnapName can be empty?

Reply:
for better readability, I would like to split `getOrigSnapName` into two functions one for check the NamespacedType if its trash then gets its original name.

Reply:
Ok, if I split it into functions rest all of your comments will be addressed :) 
Onlt draw back in doing that is image open/close operation needs to be done twice.

Reply:
that would be fine as this operation happens only when the snapshot limit is reached the configuration value.